include/server/SC_WorldOptions.h

```diff
@@ -29,74 +29,62 @@
 
 typedef int (*PrintFunc)(const char *format, va_list ap);
 
-struct WorldOptions
+const struct WorldOptions
 {
-	const char* mPassword;
-	uint32 mNumBuffers;
-	uint32 mMaxLogins;
-	uint32 mMaxNodes;
-	uint32 mMaxGraphDefs;
-	uint32 mMaxWireBufs;
-	uint32 mNumAudioBusChannels;
-	uint32 mNumInputBusChannels;
-	uint32 mNumOutputBusChannels;
-	uint32 mNumControlBusChannels;
-	uint32 mBufLength;
-	uint32 mRealTimeMemorySize;
+	const char* mPassword = nullptr;
+	uint32 mNumBuffers = 1024;
+	uint32 mMaxLogins = 64;
+	uint32 mMaxNodes = 1024;
+	uint32 mMaxGraphDefs = 1024;
+	uint32 mMaxWireBufs = 64;
+	uint32 mNumAudioBusChannels = 1024;
+	uint32 mNumInputBusChannels = 8;
+	uint32 mNumOutputBusChannels = 8;
+	uint32 mNumControlBusChannels = 16384;
+	uint32 mBufLength = 64;
+	uint32 mRealTimeMemorySize = 8192; // in kilobytes
+
+	int mNumSharedControls = 0;
+	float *mSharedControls = 0;
+
+	bool mRealTime = true;
+	bool mMemoryLocking = false;
+
+	const char *mNonRealTimeCmdFilename = nullptr;
+	const char *mNonRealTimeInputFilename = nullptr;
+	const char *mNonRealTimeOutputFilename = nullptr;
+	const char *mNonRealTimeOutputHeaderFormat = nullptr;
+	const char *mNonRealTimeOutputSampleFormat = nullptr;
 
-	int mNumSharedControls;
-	float *mSharedControls;
-
-	bool mRealTime;
-	bool mMemoryLocking;
-
-	const char *mNonRealTimeCmdFilename;
-	const char *mNonRealTimeInputFilename;
-	const char *mNonRealTimeOutputFilename;
-	const char *mNonRealTimeOutputHeaderFormat;
-	const char *mNonRealTimeOutputSampleFormat;
-
-	uint32 mPreferredSampleRate;
-	uint32 mNumRGens;
+#if defined(_WIN32)
+#	define DEFAULT_PREFERRED_SAMPLE_RATE 44100
+#else
+#	define DEFAULT_PREFERRED_SAMPLE_RATE 0
+#endif
```

Comment:
this introduces a macro unnecessarily for all clients who include this header, plus doing it in the middle of the struct definition is icky. two options would be to do it without the macro name and just wrap the line below in #ifdef (which is OK by me) or to move this up above the struct definition and then #undef the macro after it's used. I think the first option is more in line with the spirit of what you're doing. I like that the definition is close to where it's used, just think that you don't need to go to all this trouble